hack/tests/sanity-check.sh

```diff
@@ -14,6 +14,4 @@ make install
 go run ./hack/generate/samples/generate_all.go
 
 # Make sure repo is still in a clean state.
-# Note that we are ignoring helm manifests with roles.
-# More info: https://github.com/operator-framework/operator-sdk/issues/3873
-git diff --exit-code -- . ':!testdata/helm/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml' ':!testdata/helm/memcached-operator/config/rbac/role.yaml'
+git diff --exit-code
```

Comment:
So, if we adopted that as a role "No exported functions should be added to the SDK tool" then, I also think that we ought to add a check in the sanity.

Reply:
Yeah I thought about that, too and I totally agree. But I think we can solve that in a follow-up.

We can use my [go-apidiff tool](https://github.com/joelanford/go-apidiff), similar to what controller-runtime is doing to check for API changes.

Reply:
Hi @joelanford, 

Sure. it shows very nice. 🥇 
  
WDYT about donate the https://github.com/joelanford/go-apidiff to this org? 
IHMO would be better for SDK/controller-runtime to use a project that at least is not personal repo in order to ensure that it will be kept maintained.